### DISCUSSION TITLE
Request to review HumanMadeObject DRAFT output

### DISCUSSION THREAD

**Discussion by beaudet** [linked-art/linked.art#731] [Q&A]
[revised_object_example.json](https://github.com/linked-art/linked.art/files/14454421/revised_object_example.json)

I'd very much appreciate a detailed review of the attached HumanMadeObject representation for compliance with the LA specs.  Thanks!

**Comment by azaroth42** [comment]
* @context, id, type, _label are good
* `equivalent` to wikidata ... This is a bit picky, but in the data space, the URI is "http://www.wikidata.org/entity/" :(. /wiki is the equivalent of /page/ in the getty vocabs
* `identified_by` -- I would add 'Primary Name' along to Titles to give a priority for the cases when there's multiple names
* `classified_as` -- sweeping and domestic activities seem like things being depicted in the image rather than physical classifications
* attribute assignments on production parts: (per #572) otherwise good
* member_of -- good
* current loc, current custodian, current owner -- good, but current owner should probably be a list
* dimension -- good
* Some statements have a `used_for` to a publishing ... what's that for? Are these citations?
* `representation` -- yep. If you want to go one further, you could just link directly to the image URL with `access_point` rather than having a record for the DigitalObject separately.

Custom metadata block -- I would put _ at the beginning of custom metadata to make it more obviously not semantic.

Custom print URL -- that could be in the data in `subject_of` for a link to a web page.

**Comment by beaudet** [comment]
- equivalent: thanks, I'll fix that
- classified_as -> yes, definitely - will split these out so that  IAD Object Name, Object Type, Style, Theme, Keywords are applied to the visual item rather than the physical object.  The other concepts should be appropriate for the piece as physical object. The IAD Object Name and Object Type are the same concept - or at least they have an equivalent to each other even if they have separate concept URIs. 
- yes, the creator roles and display order will be changed per the outcome of discussion on #572
- current_owner will be changed to a list even though it will only ever have one member
- the used_for is to convey the publication date of cited works in the bibliography for the relevant entities - the year is not always provided in the TMS text types that store this data - it depends on the text type - so when provided, I needed a way to represent the date of publication.
- regarding representation, I think there are at least two scenarios:

1. images that directly depict the (I assume visible light) generally understood notion of the visual content of the art object.  We call those images "primary images" at the NGA and I think this would include views around a sculpture in addition to the head-on photo of a 2D work.  These are not multiple copies of the same image, but all such images depict the "visual content" of the image.  My read of the visual item notes in CRM is that the decision of where to draw the line between multiple instances of visual content for a work is up to the publisher of the data but if there's guidance beyond that, it would be good to discuss / document.

2. images that are related in some way to the art object such as a comparative image that has helped with scholarly research.  Keeping track of specific visual items in that context is much more complicated so my intention is to use blank nodes for those visual items and to associated the image not through representation but through another property.  Other examples include non-visible light imagery.  This scenario is posed as a question in #576.

_custom_metadata - yup will ensure it has a leading underscore

custom print URLs - good call.  I'll make that change as well.  

Thanks - just need #576 answered to have solutions for all of it I think, then I'll regenerate a corrected sample and post.

**Comment by beaudet** [comment]
For the custom print URL, any guidance on how best to assemble a custom concept to convey the notion in a consistent manner that a digital object is a web page related to https://vocab.getty.edu/aat/300257455 (made-to-order) https://vocab.getty.edu/aat/300375334 (giclee prints)?  For example, would I just assign all three AAT terms as meta-types and call it a day?  

And is there a procedure by which the meaning of concepts can be unambiguously understood by interpreting their meta-types?  For example, is there just an analogy to multiple-inheritance?  If a concept "web page for ordering giclee prints" is minted and its metatypes are "web page", "made-to-order", and "giclee prints" - it would inherit aspects of all three types.  Are there multiple interpretations to what such a concept might mean, particularly when interpreted by automated systems that might not (yet) have the ability to use the _label to guide the interpretation.

**Comment by azaroth42** [comment]
Is there an expectation of what a consuming application would do with the classifications beyond "web page" ? I might classify it as "commercial" or similar to hint that it's a place to buy something? And the client/viewer might use a "$" icon to distinguish it from the CMS page or something like that?

Interpreting multiple types -- they each apply independently. It's a web-page, and a made-to-order, and a giclee-print... which is semantic queasiness I think in the same way as role vs technique.

**Comment by beaudet** [comment]
Our specific use case at the NGA is (and still is I think?) include a link on art object pages to this service (managed by our shops) when a print can be ordered for an art object.  These are "made to order" since there are framing and size options so keeping inventory in stock would be prohibitively expensive (I'm just assuming these points for purpose of this meta-type example). 

In a broader context and as an entrepreneur, I might want to create a service that provides a search across all such giclée printing services and re-sell the products at a slight markup or otherwise participate in an affiliates program to get a slice of the revenue.  So, it would be useful to be able to harvest services that are described using either this meta-typed term specifically or what I assume must also be expressible directly against the HumanMadeObject via some combination of concepts similarly arranged.

**Comment by beaudet** [comment]
Yes, agree on the semantic queasiness.  I think it's important to understand the limitations, what type of knowledge about the object is reasonable to assume given the applied concepts, and what type of knowledge should not be assumed. 

Considering your comment that each concept applies independently, it sounds to me like an algorithm to interpret meaning could something like this:

* for each depth=1 Concept c by which an object is classified:
        gather first listed meta-type or parent concept for each c (should we grab both?)
                the object is an instance of class c specifically and therefore also has superclass of c's meta-type (what about the parent concept of c?)

* creating a new instance of Type t and assigning an id to it, then classifying object o with t is essentially the shortened form of exploding the definition of t into all objects needing to be classified with t and it also makes t reusable.

* therefore, if one defines an interface i that all Type t's must provide an implementation of, then object o, when consuming instances of Concept, would know how to participate in an application that treats Type with object orientation in mind, for example, an icon representing each classification down to a certain level or a rendering algorithm for displaying the face of object o in the context of Concept c.  The process of these behaviors can start at the top of a given thesaurus and move down the tree to provide for more customized behaviors.

* there exists a set, s1 of all instances of Type c1.  If object o is classified as Types c1 and c2, then object o is within the subset of instances defined by the intersection of s1 and s2.  Presumably, that includes everything anyone can imagine so long as it satisfies these constraints.  So, there is certainly some wiggle room since imagination is vast, but if we state that it represents "the most obvious" solution for that intersection AND we maintain a community adopted repository for those instances, we will have made significant progress.

* there's also the question of how to handle equivalents. I think of equivalents as the loosest equality operator for types returning true.  So, more like a sort of the same thing (~) operator.  Kind of like identical twins where there are two instances of the same blueprint yet they remain distinct in all respects.  Does equivalent also imply a desire to hand-off processing to the referenced vocabulary as if it were a soft link in a file system rather than a sub-directory or is it a soft link that lives side-by-side the directory?  I think the later.  So, I guess the way to unwind equivalent is by treating the situation the same as if the equivalent Concept has been applied directly to object o?  Is anything else implied?

Thoughts on all this?

**Comment by beaudet** [comment]
Going back to the example of Giclee - the Type "Web site for ordering made-to-order prints" would not literally be:
1. a web site
2. and a giclee print
3. and is made to order

but it would primarily be a web site - that's the obvious answer so I would say that ordering of the concepts matters and the algorithm above won't work by just assigning the types literally to the object.  Also, assigning all types to object o produces a union not an intersection of behaviors.   

Yes, it's a web site.  That is the primary meta-type.  The secondary meta-types further refine the interpretation but they don't necessary impart behaviors to the object.  The web site isn't suddenly going to have a printed surface, for example. 

So it's a web site.  Obviously the web site is not a giclee print itself and it wasn't made-to-order just for me as a web site experience, but it is a web site that is itself about giclee prints and about making stuff to order, so I would argue that a reasonably intelligent interpretation is that this is a web site for ordering giclee prints through a made-to-order process, but you're right that I'm missing the commercial aspect and there should be a term for that.  I'm also not a valid interpreter for this term because I made it up, so let me ask Chat GPT 3.5 what it thinks.  And, as crazy as it it might sound, using multiple AIs to arbitrate this might actually provide a useful off-line interpreter for accelerating the custom typing process while coalescing around a standard for measurement.  I wouldn't be opposed to recommending the use of AI as a best practice for such arbitration. 

![image](https://github.com/linked-art/linked.art/assets/77699/e7ce1c1d-d710-4de8-88a9-ee2bdd22d849)

so, I started a new 3.5 chat and asked:
![image](https://github.com/linked-art/linked.art/assets/77699/7b32f502-6e05-4eae-b92d-0972e02e15fe)

Seems pretty promising, but still wondering a bit how non-AI systems can interpret these combinations of Concepts to more accurate interpret the type when it's encountered. 

similar response with GPT 4
![image](https://github.com/linked-art/linked.art/assets/77699/02a84c96-1e6c-42e8-8550-1ea51c733e50)

**Comment by beaudet** [comment]
In my experimentation, it's probably not a surprise that the ordering of the meta-types as interpreted by LLMs follows sentence construction a bit.  I added *selling as a term and then applied an arbitrary concept like * mathematician and tested the responses with mathematician at different spots in the line-up and got different interpretations.  So, I think this is a pretty straightforward and reasonable tool to help us settle the interpretation question to narrow down to the most obvious combined concept. 

But I do think the order of terms matters very much to the resulting concept.

**Comment by beaudet** [comment]
[girl_with_a_broom_draft_2.json](https://github.com/linked-art/linked.art/files/14565611/girl_with_a_broom_draft_2.json)


Here's an updated version of Girl with a Broom as Linked Art. Eyeballs appreciated.

(github links are to an internal repository while in development and useful during development until a URI resolution service is deployed.

**Comment by beaudet** [comment]
closing this particular discussion as the json referenced has been updated since this review was requested.
